allennlp/data/token_indexers/pretrained_transformer_pretokenized_indexer.py

```diff
@@ -0,0 +1,95 @@
+from typing import Dict, List, Tuple
+import logging
+
+from overrides import overrides
+import torch
+
+from allennlp.common.util import pad_sequence_to_length
+from allennlp.data.vocabulary import Vocabulary
+from allennlp.data.tokenizers.token import Token
+from allennlp.data.token_indexers import PretrainedTransformerIndexer, TokenIndexer
+from allennlp.data.token_indexers.token_indexer import IndexedTokenList
+from allennlp.data.tokenizers import PretrainedTransformerTokenizer
+
+logger = logging.getLogger(__name__)
+
+
+@TokenIndexer.register("pretrained_transformer_pretokenized")
```

Comment:
Ok, I was thinking "pretokenized" as in already tokenized into words but not wordpieces. I agree that it's not very accurate. But I think `Mismatched` might be a little generic and confusing. What is mismatched here? The indexer input and the transformer model input? Many other things could cause this mismatch though, e.g. I'm implementing a long sequence folding logic which splits (indexer input) [CLS] A B C D E F [SEP] into (embedder input) [ [CLS] A B C [SEP], [CLS] D E F [SEP] ]. This is also "mismatched" in a sense.

Reply:
"mismatched" here refers to mismatched tokenization and modeling. I think that accurately describes the problem and gives a hint to the reader about why this is more complicated case with harder logic, and that they might want to think twice about using it (because most use cases probably shouldn't use this anymore).  If we use that name, I would state this explicitly in the docstring.

Reply:
Ok let's used "mismatched" for now. I don't know whether "most use cases probably shouldn't use this" is true though. This should be applicable to many span- or token-based tasks (as opposed to sentence-based), e.g. NER, coref, QA, parsing, etc.

Reply:
The reason I say most use cases probably shouldn't use this is because you're very likely better off distributing your labels instead of having mismatched tokenization.

Reply:
Though I admit that at this point this is just intuition, not something with hard empirical evidence.